src/server/dfly_main.cc

```diff
@@ -701,6 +711,46 @@ void PrintBasicUsageInfo() {
   std::cout << endl;
 }
 
+void ParseFlagsFromEnv() {
+  if (getenv("DFLY_PASSWORD")) {
+    LOG(WARNING)
+        << "DFLY_PASSWORD environment variable is being deprecated in favour of DFLY_requirepass";
+  }
+  // Allowed environment variable names that can have
+  // DFLY_ previx, but don't necessirily have an ABSL flag created
+  std::set<std::string_view> allowed_environment_flag_names = {"dev_env", "password"};
+  const auto& flags = absl::GetAllFlags();
+  for (char** env = environ; *env != nullptr; env++) {
+    constexpr string_view kPrefix = "DFLY_";
+    string_view environ_var = *env;
+    if (absl::StartsWith(environ_var, kPrefix)) {
+      // Per 'man environ', environment variables are included with their values
+      // in the format "name=value". Need to strip them apart, in order to work with flags object
+      pair<string_view, string_view> environ_pair = absl::StrSplit(
+          absl::AsciiStrToLower(absl::StripPrefix(environ_var, kPrefix)), absl::MaxSplits('=', 1));
```

Comment:
Sorry to have missed this before, but why are you lowercasing? Is it in order for users to be able to specify for example `DFLY_REQUIREPASS` instead of `DFLY_requirepass`?
I actually wouldn't do that, but this is more severe as you're also lowercasing the value, so values won't be able to have uppercase strings which is not something we want
I'd drop the lowercasing altogether unless it's necessary for some reason I didn't think of

Reply:
Huh, looking again, this is actually unsafe (undefined behavior)
The reason is that `AsciiStrToLower()` returns a `std::string` (i.e. by value), and then we're taking `string_view`s to point into that tmp value after it's gone

Reply:
I see! Basically, it was just left from the previous logic since there was a mismatch between cases for FLAG-related env vars and the ones created manually